rfc/rfc-78/rfc-78.md

```diff
@@ -0,0 +1,301 @@
+- hoodie.record.merge.mode
+- New metadata partitions could be added (optionally enabled)
+
+### MDT changes
+- New MDT partitions are available in 1.x. MDT schema upgraded.
+- RLI schema is upgraded to hold row positions.
+
+### Timeline:
+- [storage changes] Completed write commits have completed times in the file name.
+- [storage changes] Completed and inflight write commits are in avro format which were json in 0.x.
+- We are switching the action type for pending clustering from “replace commit” to “cluster”.
+- [storage changes] Timeline ➝ LST timeline. There is no archived timeline in 1.x.
+- [In-memory changes] HoodieInstant changes due to presence of completion time for completed HoodieInstants.
+
+### Filegroup/FileSlice changes:
+- Log file names contain delta commit time instead of base instant time.
+- Log appends are disabled in 1.x. In other words, each log block is already appended to a new log file.
+- File Slice determination logic for log files changed. In 0.x, we have base instant time in log files and its straight forward. 
+In 1.x, we find completion time for a log file and find the base instant time (parsed from base files) for a given HoodieFileGroup, 
+- which has the highest value lesser than the completion time of the log file) of interest. 
+- Log file ordering within a file slice. (in 0.x, we use base instant time ➝ log file versions ➝ write token) to order diff log files. in 1.x, we will be using completion time to order).
+- Rollbacks in 0.x appends a new rollback block (new log file). While in 1.x, rollback will remove the partially failed log files. 
+
+### Log format changes:
+- We have added new header types in 1.x. (IS_PARTIAL)
+
+## Changes to be ported over to 0.16.x to support reading 1.x tables
+
+### What will be supported
+- For features introduced in 0.x, and tables written in 1.x, 0.16.0 reader should be able to provide consistent reads w/o any breakage.
+- 
+### What will not be supported
+- A 0.16 writer cannot write to a table that has been upgraded-to/created using 1.x without downgrading to 0.16. Might be obvious, but calling it out nevertheless.
+- For new features introduced in 1.x, we might call out that 0.16.x reader may not support reads.
+
+| 1.x Features written by 1.x writer        | 1.x reader   | 0.16.x Reader | 
+|----------------|-------------------------------------------|-------|
+| Deletion vector | supported | Falls back to key based merges giving up on perf optimization
+| Partial merges/updates  | supported | Fails with clear error message stating that partial merging |
+| Functional indexes | supported | Not supported. Perf optimization may not kick in|
+| Secondary indexes | supported | Not supported. Perf optimization may not kick in|
+| NBCC OR Completion time based log file ordering in a file slice | supported | Not supported. Will be using log file version and write token based ordering.|  
+
+
+### Timeline
+- Timeline read of 1.x need to be supported.
```

Comment:
Our idea here is to not port LSM archived timeline reader to 0.16.x.
We need to fix 1.x timeline to align uncleaned timeline as active and rest of it as archived or a minimum of 2 days of timeline. 

For eg, 
Case 1: Someone configures cleaner to 4 days. 

1.x active timeline will be 4 days and rest will go into LSM 

Case2: Someone configures cleaner to 1 day:
1.x active timeline will be 2 days and rest will go into LSM.

If we maintain this semantics, with 0.16.x reader, we don't need to port LSM reader. just reading active timeline would suffice for all of snapshot, time travel and incremental queries.